src/volttron/utils/dynamic_helper.py

```diff
@@ -52,28 +52,40 @@ def get_klass(module: str | ModuleType, klass_name: str) -> Type:
         raise e
 
 
-def get_subclasses(module: ModuleType, parent_klass: Type, return_all=False) -> List[Type]:
+def get_subclasses(module: ModuleType | str,
+                   parent_class: Type | str,
+                   return_all=False) -> List[Type]:
     """Returns a list of subclasses of a specific type. If return_all is set to False,
     returns all subclasses. Default behavior is to return the first subclass found in the module.
 
     :param module: A module containing classes
-    :type module: ModuleType
-    :param parent_klass: The parent class that could be a parent of classes in the module
-    :type parent_klass: Type
+    :type module: ModuleType | str
+    :param parent_class: The parent class that could be a parent of classes in the module
+    :type parent_class: Type | str
     :param return_all: True if all subclasses are desired; False if only the first subclass. Defaults to False
     :type return_all: bool, optional
-    :raises ValueError:
+    :raises ValueError: Raises ValueError if no subclasses are found.
     :return: A list of sublcasses of a specific type
     :rtype: List
     """
     all_subclasses = []
-    for c in inspect.getmembers(module, inspect.isclass):
-        if parent_klass in c[1].__bases__:
+    module_temp, parent_class_temp = _check_module_parent_class(module, parent_class)
```

Comment:
Unless we are doing it in multiple places you don't need the check_module_parent_class function...

just check if module is a string and use get_module if it is...then loop over the attributes of the module.

Why do you need a tuple return value?